packages/channel/src/insights-helper.ts

```diff
@@ -0,0 +1,95 @@
+import { type CurrencyEnum, type DeviceEnum, type Insight, prisma, type PublisherEnum } from '@repo/database';
+import { AError, FireAndForget, groupBy as groupByUtil } from '@repo/utils';
+import * as changeCase from 'change-case';
+import {
+  type FilterInsightsInputType,
+  type InsightsColumnsGroupByType,
+  type GroupedInsightsWithEdges,
+} from '@repo/channel-utils';
+import { getInsightsCache, setInsightsCache } from './insights-cache';
+import { groupedInsights } from './insights-query-builder';
+
+const fireAndForget = new FireAndForget();
+
+export interface FieldNode {
+  kind: string;
+  name: {
+    value: string;
+  };
+  selectionSet?: {
+    selections: {
+      kind: string;
+      name: { value: string };
+    }[];
+  };
+}
+
+export const getInsightsHelper = async (
+  filter?: FilterInsightsInputType, // TODO: should be required
+  organizationId?: string, // TODO: should be required
```

Comment:
```suggestion
  filter: FilterInsightsInputType,
  organizationId: string,
```
let's make the mandatory from now.